pkg/webhook/vault_secret_manager.go

```diff
@@ -74,8 +74,9 @@ func (i VaultSecretManagerInjector) Inject(ctx context.Context, secret *coreIdl.
 			return p, false, err
 		}
 
-		p.ObjectMeta.Annotations = utils.UnionMaps(p.ObjectMeta.Annotations, commonVaultAnnotations)
-		p.ObjectMeta.Annotations = utils.UnionMaps(p.ObjectMeta.Annotations, secretVaultAnnotations)
+		p.ObjectMeta.Annotations = utils.UnionMaps(i.cfg.ExtraAnnotations, p.ObjectMeta.Annotations)
```

Comment:
Note that I change the ordering of `utils.UnionMaps` arguments so that user-defined annotations are given priority. This allows overriding annotation created by the webhook.
For instance, in my org, we have a multi-tenant Vault where we need a specific `role` to access a secret. However, Flyte <> Vault integration can only support single role. Thus, to use different role user can do so by specifying annotation in their workflow
```
@task(
    secret_requests=[
        Secret(
            group=SECRET_GROUP,
            key=SECRET_NAME,
            mount_requirement=Secret.MountType.FILE,
        )
    ],
    pod_template=PodTemplate(annotations={
        "vault.hashicorp.com/role" : ROLE,
    })
)
```